node-graph/graph-craft/src/proto.rs

```diff
@@ -378,8 +425,30 @@ mod test {
 		assert_eq!(construction_network.nodes[5].1.construction_args, ConstructionArgs::Nodes(vec![3, 4]));
 	}
 
+	#[test]
+	fn stable_node_id_generation() {
+		let mut construction_network = test_network();
+		construction_network.reorder_ids();
+		construction_network.generate_stable_node_ids();
+		construction_network.resolve_inputs();
+		construction_network.generate_stable_node_ids();
+		assert_eq!(construction_network.nodes[0].1.identifier.name.as_ref(), "value");
+		let ids: Vec<_> = construction_network.nodes.iter().map(|(id, _)| *id).collect();
+		assert_eq!(
+			ids,
+			vec![
+				17495035641492238530,
+				14931179783740213471,
+				2268573767208263092,
+				14616574692620381527,
+				12110007198416821768,
+				11185814750012198757
```

Comment:
> The internal algorithm is not specified, and so it and its hashes should not be relied upon over releases.

From std::collections::hash_map::DefaultHasher docs. Perhaps we could test another way to avoid the test breaking in new rust versions.

Reply:
That is true but the test is not that helpful anyway because I never actually checked the ids are correct, it is mainly there to detect changes. Not sure how to best test the desired behavior. I guess we could compare the nodes for different node networks to check if it behaves correctly